orttraining/orttraining/core/optimizer/megatron_transformer.cc

```diff
@@ -1212,10 +1209,10 @@ Status MegatronTransformer::TransformBARTAttention(Graph& graph, bool& modified,
                                     sa_f_input_defs,
                                     {&sa_f_out_arg}, {}, kMSDomain);
     sa_f_node.SetExecutionProviderType(k_matmul_ptr->GetExecutionProviderType());
-    graph_utils::ReplaceNodeInput(*k_matmul_ptr, 0, *(sa_f_node.MutableOutputDefs()[0]));
-    graph_utils::ReplaceNodeInput(*v_matmul_ptr, 0, *(sa_f_node.MutableOutputDefs()[0]));
+    ReplaceNodeInput(graph, *k_matmul_ptr, 0, sa_f_node, 0, FilterConsumeNodeWithOpType);
```

Comment:
will this trigger ReplaceDownstreamNodeInput in any scenario? If yes, can we add a test for that?